src/DynamoCoreWpf/ViewModels/Preview/InfoBubbleViewModel.cs

```diff
@@ -275,27 +297,61 @@ private void GenerateContent()
                     Content = FullContent;
                     break;
             }
-        }        
-
+        }
         #endregion
     }
 
     public struct InfoBubbleDataPacket
     {
+        private const string externalLinkIdentifier = "href=";
         public InfoBubbleViewModel.Style Style;
         public Point TopLeft;
         public Point BotRight;
         public string Text;
+        public Uri Link;
         public InfoBubbleViewModel.Direction ConnectingDirection;
 
-        public InfoBubbleDataPacket(InfoBubbleViewModel.Style style, Point topLeft, Point botRight,
-            string text, InfoBubbleViewModel.Direction connectingDirection)
+        public InfoBubbleDataPacket(
+            InfoBubbleViewModel.Style style,
+            Point topLeft,
+            Point botRight,
+            string text,
+            InfoBubbleViewModel.Direction connectingDirection)
         {
             Style = style;
             TopLeft = topLeft;
             BotRight = botRight;
+            Link = CheckIfMessageHasHrefLinkAndExtract(ref text);
             Text = text;
             ConnectingDirection = connectingDirection;
         }
+
+
+        //Check if has link
+        private static Uri CheckIfMessageHasHrefLinkAndExtract(ref string text)
```

Comment:
why are you using a ref? So you can modify the text? ... advise considering if this is the best design, I see that even after modifying the text to remove the link the url creation can still fail... so thats a bit weird.

might make more sense to split into two functions and return a bool from the first one - ie `CheckIfMessageHasHref` and `CreateURIFromMessageText`

Reply:
the link creation being able to fail after sanitising the text was on purpose, cause you could pass in `text href=` and the text is fine but there's no link to parse. Was trying to avoid duplicate work but agree it's not cleanest design, have split them up now, with the caveat that it's slightly less efficient.